scripts/run_nodes.sh

```diff
@@ -1,123 +1,289 @@
+# validators have ids //N, //N+1, ...
+# Obviously, this is just for testing reasons and such seeds must be never used in production.
+# Also, each AlephNode chain has sudo account. Here we assume it is //Alice.
+#
+# Make sure a machine on which you're running your script has enough RAM memory. For testing nodes
+# with empty db, a 1.5 GB per node would be enough.
+#
+# They are 3 set of ports you need to have opened and free on your machine in order to run consensus:
+#  * RPC port - range [9944; 9954) - used for JSON RPC protocol
+#  * P2p port - range [30333; 30343) - used for P2P peer network
+#  * Validator port - range [30343; 30353) - used for consensus mechanism (AlephBFT)
+#
+# You need to have installed following prerequisites in order to use that script:
+#   * jq
+#
+# This script also accepts env variables instead arguments, see --help for details. All arguments
+# are optional.
+
+set -euo pipefail
+
+# ------------------------ constants --------------------------------------
+
+export ALEPH_NODE="target/release/aleph-node"
+NODE_P2P_PORT_RANGE_START=30333
+NODE_VALIDATOR_PORT_RANGE_START=30343
+NODE_RPC_PORT_RANGE_START=9944
+
+# ------------------------ argument parsing and usage -----------------------
+
+source ./scripts/common.sh
 
 function usage(){
-  echo "Usage:
-      ./run_nodes.sh [-v N_VALIDATORS] [-n N_NON_VALIDATORS] [-b false] [-p BASE_PATH] [-l N_LISTENERES] [-r false] [ALEPH_NODE_ARG]...
-  where 2 <= N_VALIDATORS <= N_VALIDATORS + N_NON_VALIDATORS + N_LISTENERES <= 10
-  (by default, N_VALIDATORS=4, N_NON_VALIDATORS=0, N_LISTENERES=0 and BASE_PATH=/tmp)"
+  cat << EOF
+Usage:
+   $0
+    [-v|--validators VALIDATORS]
+      number of validators to bootstrap and start
+    [-n|--rpc-nodes RPC_NODES]
+      number of RPC nodes to bootstrap and start
+    [-p|--base-path BASE_PATH]
+        if specified, use given base path (keystore, db, AlephBFT backups)
+        if not specified, base path is ./run-nodes-local
+    [--dont-bootstrap]
+      set if you don't want to bootstrap chain, ie generate keystore and chainspec
+    [--dont-build]
+       set if you do not want to build testing aleph-node binary
+    [--dont-delete-db]
+      set to not delete database
+    [--dont-remove-abtf-backups]
+      set to not delete AlephBFT backups; by default there are removed since
+      this script is intended to bootstrap chain by default, in which case you do not want to have
+      them in 99% of scenarios
+EOF
+  exit 0
 }
 
-N_VALIDATORS=4
-N_NON_VALIDATORS=0
-N_LISTENERES=0
-BUILD_ALEPH_NODE='true'
-BASE_PATH='/tmp'
-BOOTSTRAP='true'
-
-while getopts "v:n:b:p:l:r:" flag
-do
-  case "${flag}" in
-    v) N_VALIDATORS=${OPTARG};;
-    n) N_NON_VALIDATORS=${OPTARG};;
-    b) BUILD_ALEPH_NODE=${OPTARG};;
-    p) BASE_PATH=${OPTARG};;
-    l) N_LISTENERES=${OPTARG};;
-    r) BOOTSTRAP=${OPTARG};;
-    *)
+VALIDATORS=${VALIDATORS:-6}
+RPC_NODES=${RPC_NODES:-1}
+BASE_PATH=${BASE_PATH:-"./run-nodes-local"}
```

Comment:
Please use relative address with respect to the script location instead of `./`, or assert before creating any backup etc., that the user is in main `aleph-node` directory

Reply:
Good point, fixed